src/convertToObject.js

```diff
@@ -1,12 +1,68 @@
 'use strict';
 
-/**
- * @param {string} sourceString
- *
- * @return {object}
- */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString === '') {
+    return {};
+  }
+
+  const firstObrabotka = sourceString;
+
+  const bigArr = firstObrabotka.split(';');
+
+  const arrFinal = bigArr.filter((str) => str.trim() !== '');
```

Comment:
You can chain the 'filter' and 'split' operations to make the code more readable and concise.